src/main/java/online/book/store/controller/BookController.java

```diff
@@ -28,30 +29,35 @@ public class BookController {
 
     private final BookService bookService;
 
+    @PreAuthorize("hasRole('ROLE_USER')")
     @GetMapping
     @Operation(summary = "Get all books", description = "Get a list of all available books")
     public List<BookDto> findAll(Pageable pageable) {
         return bookService.findAll(pageable);
     }
```

Comment:
Users with `admin ` role should have access to `findAll()` method. 
You can use @PreAuthorize annotation with parameters "hasAnyAuthority('ROLE_USER', 'ROLE_ADMIN')"
`@PreAuthorize("hasAnyAuthority('ROLE_USER', 'ROLE_ADMIN')")`

Reply:
So we can implement role-based access in several ways, but this way seems more flexible)